matchbox/resource_profile.go

```diff
@@ -193,5 +236,9 @@ func containerLinuxConfig(d *schema.ResourceData) (filename, config string) {
 		return fmt.Sprintf("%s.ign", name), content.(string)
 	}
 
+	if content, ok := d.GetOk("generic"); ok {
+		return fmt.Sprintf("%s.generic", name), content.(string)
```

Comment:
Admittedly, choosing a name on the user's behalf seems inevitable, because most won't care to provide one and it'll be viewed as an internal detail. Perhaps we just use the name outright, no suffix needed.